### ISSUE TITLE
Support for storage blob last accessed time management policy

### ISSUE THREAD

**Issue by elliot-resdiary** [hashicorp/terraform-provider-azurerm#8940]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

As of last month the 'last accessed time' tracking for blobs is available, along with policies to automatically change blob storage tiers after 'last accessed time' reaches a certain value. These policies also support changing a blob's tier from cool to hot immediately if it's accessed.

https://azure.microsoft.com/en-gb/updates/azure-blob-access-time-tracking-and-access-timebased-lifecycle-management-preview/

I think this would be a great addition to the [azurerm_storage_management_policy](https://www.terraform.io/docs/providers/azurerm/r/storage_management_policy.html) resource, which currently only supports policies based on 'last modified time'.

### New or Affected Resource(s)

* azurerm_storage_management_policy

### Potential Terraform Configuration

```hcl

resource "azurerm_storage_management_policy" "example" {
  storage_account_id = azurerm_storage_account.example.id

  rule {
    name    = "rule1"
    enabled = true
    actions {
      base_blob {
        tier_to_cool_after_days_since_access_greater_than    = 10
        tier_to_archive_after_days_since_access_greater_than = 50
        delete_after_days_since_access_greater_than          = 100
        tier_from_cool_to_hot_on_access                      = true
      }
    }
  }
}
```

**Comment by dantdj** [comment]
Hey @yupwei68 - do you know if support for this feature in this provider is on the roadmap for when the feature becomes generally available?

**Comment by yupwei68** [comment]
Hi @dantdj Yes. I'll keep in contact with the service team. Once it's globally released, it'll be supported in Terraform

**Comment by dantdj** [comment]
@yupwei68 - Thanks, that sounds great! Do you know if the releases will be roughly at the same time, or will there be a bit of a delay?

**Comment by gjrtimmer** [comment]
Any update on this, I would like to write a policy in Terraform with Last Accessed ?

**Comment by SimonGasparKentico** [comment]
It looks that azurerm 2.99.0 support this functionality https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_management_policy

**Comment by dskdas** [comment]
the above i.e. 

```

          + actions {
              + base_blob {
                  + delete_after_days_since_last_access_time_greater_than          = -1
                  + delete_after_days_since_modification_greater_than              = 5
                  + tier_to_archive_after_days_since_last_access_time_greater_than = -1
                }
            }

```

causes below error:

```

storage.ManagementPoliciesClient#CreateOrUpdate: Failure responding to request: StatusCode=400 
-- Original Error: autorest/azure: Service returned an error. Status=400
Code="MissingLastAccessTimeBasedTrackingPolicy" Message="Last access time based tracking policy
must be enabled before using its specific actions in object lifecycle management policy"

```

downgrading to 2.96.0 results in the plan reflecting below and it helps get past this issue for now.

```

          + actions {
              + base_blob {
                  + delete_after_days_since_modification_greater_than = 5
                }
            }

```

**Comment by HappyTobi** [comment]
@dskdas  you have set the following flag at your "azure_storage_account"

```
resource "azurerm_storage_account" "storage_account" {
....
blob_properties {
last_access_time_enabled = true
}
}
```

After adding the `last_access_time_enabled` property everything should work fine.

See also the provider tests:
https://github.com/hashicorp/terraform-provider-azurerm/blob/117cdd4dce11a88e813bda7e46b40f16b03c5340/internal/services/storage/storage_management_policy_resource_test.go#L992

**Comment by carljonpope** [comment]
Is there any way to configure the policy to move back to hot if accessed?
![image](https://user-images.githubusercontent.com/49517856/192984847-b6e930cd-ab66-4b57-899c-de0b3f11f3e1.png)

**Comment by yriveiro** [comment]
> Is there any way to configure the policy to move back to hot if accessed? ![image](https://user-images.githubusercontent.com/49517856/192984847-b6e930cd-ab66-4b57-899c-de0b3f11f3e1.png)

https://github.com/hashicorp/terraform-provider-azurerm/issues/20600 implements that feature

**Comment by tombuildsstuff** [comment]
Closing since this has been fixed by #20600 - thanks for the heads up @yriveiro :)
